<p>
  Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all function names match a provided regular expression.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
function DoSomething(){...}
</pre>

<h2>Compliant Solution</h2>
<pre>
function doSomething(){...}
</pre>
