<p>
  alert(...) can be useful for debugging during development, but in production mode this kind of pop-up should never be displayed.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
if(unexpectedCondition)
{
  alert("Unexpected Condition");
}
</pre>
