<p>
  Because semicolons at the ends of statements are optional, starting function call arguments on a separate line makes the code confusing and might lead to errors.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
var fn = function () {
  //...
}

(function () { // Noncompliant
  //...
})();
</pre>
<p>
  What was the initial purpose of the developer ?

    <ol>
      <li>Was it to define a function and then to execute some unrelated code inside a closure ?</li>
      <li>Or to pass the second function as a parameter to the first one ?</li>
    </ol>
</p>
<p>
  The first option will be the one chosen by the ColdFusion interpreter.
</p>

<h2>Compliant Solution</h2>
<p>
  Either
</p>
<pre>
// define a function
var fn = function () {
  //...
}; // <-- semicolon added

// then execute some code inside a closure
(function () {
  //...
})();
</pre>
<p>
  or
</p>
<pre>
var fn = function () {
  //...
}(function () { // <-- start function call arguments on same line
  //...
})();
</pre>
<p>
  By extension and to improve readability, any kind of function call arguments should not start on new line.
</p>
